style/swift.md

```diff
@@ -1,3 +1,1037 @@
+
+```swift
+attendeeList.sort { a, b in
+  a > b
+}
+```
+
+Chained methods using trailing closures should be clear and easy to read in context. Moreover, they should go one per line. Example:
+
+```swift
+let value = numbers.map { $0 * 2 }.filter { $0 % 3 == 0 }.index(of: 90)
+
+let value = numbers
+  .map {$0 * 2}
+  .filter {$0 > 50}
+  .map {$0 + 10}
+```
+
+## Types
+
+Always use Swift's native types and expressions when available. Swift offers bridging to Objective-C so you can still use the full set of methods as needed.
+
+**Preferred**:
+```swift
+let width = 120.0                                    // Double
+let widthString = "\(width)"                         // String
+```
+
+**Less Preferred**:
+```swift
+let width = 120.0                                    // Double
+let widthString = (width as NSNumber).stringValue    // String
+```
+
+**Not Preferred**:
+```swift
+let width: NSNumber = 120.0                          // NSNumber
+let widthString: NSString = width.stringValue        // NSString
+```
+
+In drawing code, use `CGFloat` if it makes the code more succinct by avoiding too many conversions.
+
+### Constants
+
+Constants are defined using the `let` keyword and variables with the `var` keyword. Always use `let` instead of `var` if the value of the variable will not change.
+
+**Tip:** A good technique is to define everything using `let` and only change it to `var` if the compiler complains!
+
+You can define constants on a type rather than on an instance of that type using type properties. To declare a type property as a constant simply use `static let`. Type properties declared in this way are generally preferred over global constants because they are easier to distinguish from instance properties. Example:
+
+**Preferred**:
+```swift
+enum Math {
+  static let e = 2.718281828459045235360287
+  static let sqrt2 = 1.41421356237309504880168872
+}
+
+let hypotenuse = side * Math.sqrt2
+
+```
+**Note:** The advantage of using a case-less enumeration is that it can't accidentally be instantiated and works as a pure namespace.
+
+**Not Preferred**:
+```swift
+let e = 2.718281828459045235360287  // pollutes global namespace
+let sqrt2 = 1.41421356237309504880168872
+
+let hypotenuse = side * sqrt2 // what is root2?
+```
+
+### Static Methods and Variable Type Properties
+
+Static methods and type properties work similarly to global functions and global variables and should be used sparingly. They are useful when functionality is scoped to a particular type or when Objective-C interoperability is required.
+
+### Optionals
+
+Declare variables and function return types as optional with `?` where a `nil` value is acceptable.
+
+Use implicitly unwrapped types declared with `!` only for instance variables that you know will be initialized later before use, such as subviews that will be set up in `viewDidLoad()`. Prefer optional binding to implicitly unwrapped optionals in most other cases.
```

Comment:
Not sure it's a good idea to have implicitly unwrapped in `viewDidLoad()`

We should maybe swap this out for [this](https://github.com/bbc/news-app-ios/blob/dev/iOSCodingGuide.md#forced-unwrap) from the news app repo style guide.